reframe/frontend/executors/policies.py

```diff
@@ -499,5 +504,11 @@ def exit(self):
                 self._failall(e)
                 raise
 
+            if len(self.stats.failures()) >= self.max_failures:
+                exc = MaxFailError('the maximum number of failures has been '
+                                   'reached')
```

Comment:
Instead of duplicating the logic here, you could put it in the `on_task_failure()`, couldn't you?

Reply:
I don't know if it is a good idea, because it creates a loop now, since `fail` will call `_failall` , which will call `fail` for each task and since the maxfail will be reached it will try to fail all every time. And even when I try to abort them only once I still get some issues with the other lists. If you think it is much better like this, I can look into where the errors are coming from.

Reply:
No, leave it as it is. I have made another suggestion, which would make the `if` simpler.